lib/publiccloud/provider.pm

```diff
@@ -426,20 +431,23 @@ sub terraform_apply {
     my $output = decode_json(script_output("terraform output -json"));
     my $vms;
     my $ips;
+    my $resource_id;
     if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
         foreach my $vm_type ('cluster_nodes', 'drbd', 'netweaver') {
             push @{$vms}, @{$output->{$vm_type . '_name'}->{value}};
             push @{$ips}, @{$output->{$vm_type . '_public_ip'}->{value}};
         }
-    }
-    else {
+    } else {
         $vms = $output->{vm_name}->{value};
         $ips = $output->{public_ip}->{value};
+        # Extract resource ID from output. The output looks like the following:
+        $resource_id = $output->{resource_id}->{value};
```

Comment:
I will add this as requested but I don't like this.

Once again: We are adding a hardcoded conditional check with no real value. If the user does not provide this resource in the terraform file, then one would expect that the value is not set or empty. This is the case, and the [GCE](http://duck-norris.qam.suse.de/t8625) test runs shows that there are no side effects.

On the other hand, if we add another terraform profile where we also need the resource_id we then have to change this line in `provider.pm` as well. And probably a engineer gets some gray hair to figure out, why `resource_id` is not set in `instance`, although it is defined in the new terraform profile.

Adding hardcoded checks like this on components that should work with any terraform profiles is something that I don't like because they are a tripping hazard.

Reply:
IMO flexibility which you trying to introduce here is artificial . Our code from several different perspectives is tightly coupled with terraform profiles which we are using . And I don't believe in bright future where we will have code which can run with ANY random terraform profile . Even if I am missing something and it is really possible I am not sure that I really want that future . There is some contract between test code and terraform profile and it is good thing because it gives me some confidence in environment which we have . 
Azure NFS is *first* time our standard profile was not enough to cover our needs . So now we have only 2 profiles for one provider and 1 profile per provider for 2 others . If we will start getting more and more than we can discuss options how to deal with that , but IMO it is too early discussion as for today ...

Reply:
Issue resolved in discussion, closing thread to avoid unnecessary wall-of-text.